app/controllers/shop_controller.rb

```diff
@@ -44,7 +44,7 @@ def changeable_orders_alert
   private
 
   def filtered_json(products_json)
-    if applicator.send(:rules).any?
+    if applicator.rules.any?
```

Comment:
I made this one public. If we use it from outside, it's part of applicator's public API